api/api/controllers/search_controller.py

```diff
@@ -300,8 +315,49 @@ def create_search_query(
     )
 
 
-def search(
-    search_params: media_serializers.MediaSearchRequestSerializer,
+def build_collection_query(
+    search_params: MediaListRequestSerializer,
+    collection_params: dict[str, str],
+):
+    """
+    Build the query to retrieve items in a collection.
+    :param collection_params: `tag`, `source` and/or `creator` values from the path.
+    :param search_params: the validated search parameters.
+    :return: the search client with the query applied.
+    """
+    search_query = {"filter": [], "must": [], "should": [], "must_not": []}
+    # Apply the term filters. Each tuple pairs a filter's parameter name in the API
+    # with its corresponding field in Elasticsearch. "None" means that the
+    # names are identical.
+    filters = [
+        # Collection filters allow a single value.
+        ("tag", "tags.name"),
```

Comment:
Weird, while digging through our data to find examples where white-space sensitivity could matter for tags, I found some tags that are _not_ normalised to lower case:

https://api.openverse.engineering/v1/audio/03ea0149-d9c5-47f3-97db-ed7f715b46af/

That result has "Low-Quality" rather than "low-quality" which other audio results have...

Reply:
One more thing, which definitely needs a change: `tags.name` is a text field, which the ES documentation says _not_ to use a terms query on: https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-term-query.html#avoid-term-query-text-fields

We should change this to `tags.name.keyword` if we want to use a terms query.

Alternatively, if we want to avoid all the exact-matching stuff, we could use `tags.name`, but then just need to change the `term` query to a `match` query below.

```suggestion
        ("tag", "tags.name.keyword"),
```

Reply:
BTW, noticing when testing locally that case insensitivity is working. I didn't notice the call to lower the tag in the media view's implementation of the tag endpoint. Please disregard my comments about case-sensitivity not working. However, if we switch to the `match` query, then we can remove the code calling tolower on the tags.

But! Because we _do_ have tags that are not lower cased, because of #3342, I think that bug is much more significant, because if we roll this feature out as-is, then there are works tagged such that they would never appear in the tagged collection even though they have that tag. That's only relevant if we stick with a terms query for tags, though.

Reply:
I think we should stick to the terms query for tags here. I've added some comments on #3342 on proposed solution. I think we can finish it by the time the additional search views are shipped.